docs-src/docs/guides/terraform_beyond_layer0.md

```diff
@@ -0,0 +1,519 @@
+
+### Security
+A Terraform state file is written in plain text. This can lead to a situation where deploying resources that require sensitive data can result in the sensitive data being stored in the state file. To minimize exposure of sensitive data, you can enable [server side encryption](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html) of the state file by adding property `encrypt` set to `true`.
+
+This will ensure that the file is encrypted in S3 and by using a remote backend, you will also have the added benefit of the state file not being persisted to disk locally as it will only ever be held in memory by Terraform.
+
+For securing the state file further, you can also enable access logging on the S3 bucket you are using for the remote backend, which can help track down invalid access should it occur.
+
+## Part 6: Terraform Configuration Structure
+
+There are many different approaches to setup your Terraform code structure. Although there is no one prescrived approach, whatever approach you take needs to be catered for the needs of your particular project. Keeping that in mind; the file structure for [Terraform beyond Layer0 example](https://github.com/quintilesims/layer0-examples/blob/master/terraform-beyond-layer0/example-1) is as below:
+
+```
+example1/
+  ─ main.tf  
+  ─ variables.tf  
+  ─ output.tf  
+  + modules/  
+      + guestbook_service/
+        ─ main.tf  
+        ─ variables.tf  
+        ─ output.tf
+      + service2/
+      + service3/
+```
+
+Here we are making use of Terraform [Modules](https://www.terraform.io/docs/modules/index.html). Modules in Terraform are self-contained packages of Terraform configurations, that are managed as a group. Modules are used to create reusable components in Terraform as well as for basic code organization. In this example, we are using modules to separate each service and making it consumable as a module.
+
+If you wanted to add a new service, you can create new folder service folder inside the ./modules. If you wanted to you could even run multiple copies of the same service. See here for more information about [Creating Modules](https://www.terraform.io/docs/modules/create.html).
+
+When creating a module, ensure that resources you are creating are prefixed with the environment and the module's name variable to ensure your resources are unique for each layer0 environment and each reference to a module.
+
+For example
+
+```
+resource "layer0_load_balancer" "guestbook" {
+  name        = "${var.name}_guestbook_lb"
+  environment = "${var.layer0_environment_id}"
+
+  port {
+    host_port      = 80
+    container_port = 80
+    protocol       = "http"
+  }
+}
+
+# note we are prefixing with the environment name also to ensure there won't be conflicts
+# if multiple instance of this application were deployed
+resource "aws_dynamodb_table" "guestbook" {
+  name           = "${var.layer0_environment_name}_${var.name}_${var.table_name}"
```

Comment:
good point - this helped me during testing where I didn't have ensure unique names as I scaled the environments up; but from a module consumer perspective, i would expect the same too.